webdriver/tests/execute_pin/execute_pin.py

```diff
@@ -0,0 +1,61 @@
+import pytest
```

Comment:
This file should ideally be spread in separate files, one for each of webdriver/tests/execute_script/ and webdriver/tests/execute_async_script/.

You should also test pinning separately from script execution, and this file seems to do a little of both. This means you’ll have to add script pinning to the WPT webdriver client API here:
https://searchfox.org/mozilla-central/source/testing/web-platform/tests/tools/webdriver/webdriver

Please also remember to add a new error type for the `no such script` error code.